doc/contributing/adding-new-napi-api.md

```diff
@@ -1,66 +1,60 @@
-    * The API **must** be implemented in a Node.js implementation with an
-      alternate VM.
+## Core principles
 
-Since the adoption of the policy whereby moving to a later version of Node-API
-from an earlier version may entail rework of existing code, it is possible to
-introduce modifications to already-released Node-APIs, as long as the
-modifications affect neither the ABI nor the API of earlier versions. Such
-modifications **must** be accompanied by an opt-out flag. This provides add-on
-maintainers who take advantage of the initial compile-time flag to track
-impending changes to Node-API with
+1. **Adherence to Node-API standards**
+   * **Must** be a C API.
+   * **Must** not throw exceptions.
+   * **Must** return `napi_status`.
+   * **Should** consume `napi_env`.
+   * **Must** operate on primitive data types, pointers to primitive data types, or opaque handles.
+   * **Must** be a necessary API, not a convenience API (which belongs in node-addon-api).
+   * **Must** not break ABI compatibility with other Node.js versions.
 
-* a quick fix to the breakage caused,
-* a notification that such breakage is impending, and thus
-* a buffer to adoption above and beyond the one provided by the initial
-  compile-time flag.
+2. **Maintaining VM agnosticism**
+   * New APIs **should** be compatible with various JavaScript VMs.
+
+## Documentation and testing
+
+1. **Documentation**
+   * PRs introducing new APIs **must** include corresponding documentation updates.
+   * Experimental APIs **must** be clearly documented as experimental and require an explicit compile-time flag
+     to opt-in (`#define`).
+
+2. **Testing**
+   * PRs **must** include at least one test case demonstrating API usage.
+   * **Should** include test cases for various interesting uses of the API.
+   * **Should** provide a sample demonstrating realistic usage, similar to a real-world addon.
+
+## Process and approval
+
+1. **Team discussion**
+   * New APIs **should** be discussed in a Node-API team meeting.
+
+2. **Review and approval**
+   * A new API addition **must** be signed off by at least two Node-API team members.
+   * **Should** be implemented in at least one other VM implementation of Node.js.
+
+3. **Experimental phase**
+   * New APIs **must** be marked as experimental for at least one minor Node.js release before promotion.
+   * **Must** have a feature flag (`NODE_API_EXPERIMENTAL_HAS_<FEATURE>`) for distinguishing experimental features.
+   * **Must** be considered for backporting.
+   * Exit criteria from experimental status include:
+     * Opening a PR in `nodejs/node` to remove experimental status, tagged as **node-api** and **semver-minor**.
+     * Approval by the Node-API team.
+     * Availability of a down-level implementation if backporting is needed.
+     * Usage by a published real-world module to support de-experimentation.
+     * Implementation in a Node.js version with an alternate VM.
```

Comment:
I think we no longer maintain an anternative Node.js/VM like node-chakracore. This should be re-phrased like

```suggestion
     * Implementation in an alternative VM.
```

/cc @nodejs/node-api